src/main/java/com/mabl/integration/jenkins/validation/MablRestApiClientRetryHandler.java

```diff
@@ -0,0 +1,37 @@
+package com.mabl.integration.jenkins.validation;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ServiceUnavailableRetryStrategy;
+import org.apache.http.impl.client.DefaultServiceUnavailableRetryStrategy;
+import org.apache.http.protocol.HttpContext;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.apache.commons.httpclient.HttpStatus.SC_NOT_IMPLEMENTED; // 501
+import static org.apache.commons.httpclient.HttpStatus.SC_BAD_GATEWAY; // 502
+
+public class MablRestApiClientRetryHandler implements ServiceUnavailableRetryStrategy {
+    private final ArrayList<Integer> retryStatusCodes = new ArrayList<Integer>(Arrays.asList(
+            SC_NOT_IMPLEMENTED,
+            SC_BAD_GATEWAY
+    ));
+    private static final int maxRetries = 5;
+    private static final long retryInterval = 1000L;
```

Comment:
We likely want much longer here, as the API is probably being deployed, which will likely take longer than 5s. How about 6s (30s total). Could go longer, but don't want to take too long to fail either.